cybersyn/scripts/factorio-api.lua

```diff
@@ -121,7 +121,7 @@ end
 
 local condition_wait_inactive = {type = "inactivity", compare_type = "and", ticks = INACTIVITY_TIME}
 local condition_only_inactive = {condition_wait_inactive}
-local condition_unloading_order = {{type = "empty", compare_type = "and"}}
+local condition_unloading_order = {{type = "empty", compare_type = "and"}, condition_wait_inactive}
```

Comment:
That ignores the station combinator's setting completely. What about building the combinator with wait_inactive=on by default. That would still allow to opt-out.

Reply:
So, believe it or not, that combinator setting does not at all do what anyone would expect. I had the same thought, but did not want to change existing behavior.

The behavior is documented (after a fashion) here: [cybersyn-allow-cargo-in-depot=If checked, trains will be allowed to have cargo in depots; no alerts will be generated and the train will not be held. In addition, trains with orders to visit requester stations with "Inactivity condition" checked will wait for inactivity instead of waiting for empty cargo. Useful for creating train systems where depots handle excess cargo. For advanced users only.](https://github.com/mamoniot/project-cybersyn/blob/af7f54c1b2e4be9d560453cbac09df0ee312af13/cybersyn/locale/en/base.cfg#L31)

If there is consensus among the contributors I am definitely willing to rewrite this so the setting does what everyone would expect, allowing people to opt out of this. But it would technically be a breaking change.

Reply:
Oh, I suppose I didn't answer the question: no, it doesn't ignore the setting at all, if the setting is toggled on it will replace that behavior with the nearby "inactivity-only" behavior so there is no change to behavior in that case.